### PR TITLE
[Feature] Add BigDecimal Metric In QuickStart

### PR DESCRIPTION
**Pull request by antwigambrah**
Description:
This PR adds support for BigDecimal type in dimensionFieldSpecs

This PR will solve the following issue -- https://github.com/apache/pinot/issues/9380

Testing:
Added to FieldSpecTest.java

cc @xiangfu0

### CODE REVIEW

**Comment by Jackie-Jiang** [comment]
I don't think this is what #9380 refers to. Currently we don't allow big_decimal as dimension, and the purpose of #9380 is to add a big_decimal metric in the quick start

**Comment by antwigambrah** [comment]
HI @Jackie-Jiang, is it about ensuring the metricFieldSpecs support big_decimal in the schema? Testing via quickstart with a big_decimal in metricFieldSpecs didn't throw any exception as it did for he dimmensionFieldSpecs. Can you point me in the right direction in order to resolve the issue ?

**Comment by Jackie-Jiang** [comment]
You may take a look at `QuickStartBase` where we have multiple example tables. We want to add a new example table with BIG_DECIMAL columns in the schema, and people can use it to try some functions that apply to the BIG_DECIMAL columns

**Comment by antwigambrah** [comment]
> QuickStartBase

Great . I will check it out then

**Comment by antwigambrah** [comment]
Hi @Jackie-Jiang. Done with the updates. Please review

File: pinot-tools/src/main/resources/examples/batch/billing/billing_offline_table_config.json [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+    "invertedIndexColumns" : [],
+    "loadMode"  : "MMAP"
+  },
+  "tenants" : {
+    "broker":"DefaultTenant",
+    "server":"DefaultTenant"
+  },
+  "tableType":"OFFLINE",
+  "metadata": {}
+}
```
Comment (Jackie-Jiang) [inline pinot-tools/src/main/resources/examples/batch/billing/billing_offline_table_config.json:17]: (minor) Add an empty line in the end

File: pinot-tools/src/main/resources/examples/batch/billing/rawdata/billing_data.csv [hunk 2]
```diff
@@ -0,0 +1,695 @@
...
+41,0,813.61,A,0,5.62,105.28,0,3.33,22.39,114.23,136.62,Sunny,Kutch,1,Santa Clara,S,Townhouse,Paid,201905
+57,0,479.79,A,0,8.73,62.08,0,2.12,14.3,72.94,87.24,Hilario,Zulauf,0,Sunnyvale,M,Townhouse,Paid,201808
+74,0,589.02,A,0,8.73,76.22,0,2.55,17.15,87.5,104.65,Elizabeth,Ankunding,1,Palo Alto,S,Townhouse,Paid,201811
+115,0,583,B,450.32,15.46,85.93,40.62,4.26,28.67,146.27,174.94,Murphy,Krajcik,1,Sunnyvale,M,Townhouse,Paid,201809
+170,0,329.3,A,233.04,8.95,48.54,21.02,2.36,15.85,80.86,96.71,Shyanne,McGlynn,1,Mountain View,U,Townhouse,Paid,201808
+231,0,1248.19,A,0,17.45,161.52,0,5.37,36.13,184.33,220.46,Bessie,Kuhn,0,Santa Clara,U,Condominium,Paid,201812
+254,0,629.84,A,0,8.73,81.5,0,2.71,18.22,92.94,111.15,Trudie,Beier,0,Mountain View,S,Condominium,Paid,201904
+255,0,974.22,A,0,17.45,126.06,0,4.31,28.97,147.82,176.79,Jeremie,DuBuque,0,Palo Alto,U,Condominium,Paid,201901
+294,0,618.55,A,0,8.73,80.04,0,2.66,17.92,91.43,109.35,Dana,Bartell,0,Cupertino,U,Condominium,Paid,201906
+299,0,915.89,A,0,5.62,118.52,0,3.72,25.06,127.86,152.92,Sydni,Rosenbaum,0,Sunnyvale,S,Condominium,Paid,201903
```
Comment (Jackie-Jiang) [inline pinot-tools/src/main/resources/examples/batch/billing/rawdata/billing_data.csv:695]: (minor) Add an empty line in the end

**Review by Jackie-Jiang — COMMENTED** [review]
Mostly good

File: pinot-tools/src/main/resources/examples/batch/billing/rawdata/billing_data.csv (hunk 2 above)
Comment (antwigambrah) [reply pinot-tools/src/main/resources/examples/batch/billing/rawdata/billing_data.csv]: Done

File: pinot-spi/src/main/java/org/apache/pinot/spi/data/FieldSpec.java [hunk 3]
```diff
@@ -343,11 +343,11 @@ public boolean equals(Object o) {
     }
 
     FieldSpec that = (FieldSpec) o;
-    return EqualityUtils.isEqual(_name, that._name) && EqualityUtils.isEqual(_dataType, that._dataType) && EqualityUtils
-        .isEqual(_isSingleValueField, that._isSingleValueField) && EqualityUtils
-        .isEqual(getStringValue(_defaultNullValue), getStringValue(that._defaultNullValue)) && EqualityUtils
-        .isEqual(_maxLength, that._maxLength) && EqualityUtils.isEqual(_transformFunction, that._transformFunction)
-        && EqualityUtils.isEqual(_virtualColumnProvider, that._virtualColumnProvider);
+    return EqualityUtils.isEqual(_name, that._name) && EqualityUtils.isEqual(_dataType, that._dataType)
```
[the remark this answers is not in the document]
Comment (antwigambrah) [reply pinot-spi/src/main/java/org/apache/pinot/spi/data/FieldSpec.java]: Done

File: pinot-common/src/test/java/org/apache/pinot/common/data/FieldSpecTest.java [hunk 4]
```diff
@@ -155,6 +155,7 @@ public void testFieldSpec() {
     Assert.assertEquals(fieldSpec1.hashCode(), fieldSpec2.hashCode());
     Assert.assertEquals(fieldSpec1.getDefaultNullValue(), "null");
 
+
```
[the remark this answers is not in the document]
Comment (antwigambrah) [reply pinot-common/src/test/java/org/apache/pinot/common/data/FieldSpecTest.java]: Done

**Comment by antwigambrah** [comment]
Hi @Jackie-Jiang. Done with the updates. Please review

File: pinot-common/src/test/java/org/apache/pinot/common/data/FieldSpecTest.java [hunk 5]
```diff
@@ -268,18 +268,18 @@ public Object[][] provideTestFormatData() {
     entries.add(new Object[]{name, dataType, "-1:HOURS:EPOCH", granularity, true, null});
     entries.add(new Object[]{name, dataType, "0.1:HOURS:EPOCH", granularity, true, null});
     entries.add(new Object[]{
-        name, dataType, "1:HOURS:EPOCH", granularity, false,
-        new DateTimeFieldSpec(name, dataType, "1:HOURS:EPOCH", granularity)
+        name, dataType, "1:HOURS:EPOCH", granularity, false, new DateTimeFieldSpec(name, dataType, "1:HOURS:EPOCH",
```
Comment (Jackie-Jiang) [inline pinot-common/src/test/java/org/apache/pinot/common/data/FieldSpecTest.java:271]: Can you please revert this file? The changes no longer apply

**Review by Jackie-Jiang — APPROVED** [review]
LGTM otherwise

File: pinot-common/src/test/java/org/apache/pinot/common/data/FieldSpecTest.java (hunk 5 above)
Comment (antwigambrah) [reply pinot-common/src/test/java/org/apache/pinot/common/data/FieldSpecTest.java]: Done

**Comment by antwigambrah** [comment]
Hi @Jackie-Jiang. Done with the updates. Please review

File: pinot-common/src/test/java/org/apache/pinot/common/data/FieldSpecTest.java [hunk 6]
```diff
@@ -349,7 +349,7 @@ public void testSerializeDeserialize()
     // Multi-value dimension field with default null value.
     dimensionFields = new String[]{
         "\"name\":\"dimension\"", "\"dataType\":\"STRING\"", "\"singleValueField\":false",
-        "\"defaultNullValue\":\"default\""
+        "\"defaultNullValue" + "\":\"default\""
```
Comment (jackjlli) [inline pinot-common/src/test/java/org/apache/pinot/common/data/FieldSpecTest.java:352]: Same here.

**Review by jackjlli — APPROVED** [review]
Minor but LGTM

**Pull request merged**